src/gravity.js

```diff
@@ -0,0 +1,443 @@
+  var isSpriteThere =  bitsy[spriteCheck]()
+  var edgeMap = {
+    up: { coord: y, value: 0 },
+    down: { coord: y, value: bitsy.mapsize - 1 },
+    left: { coord: x, value: 0 },
+    right: { coord: x, value: bitsy.mapsize - 1 }
+  }
+  var edgeToCheck = edgeMap[dir]
+  var isWallThere = bitsy[wallCheck]() || (edgeToCheck.coord == edgeToCheck.value)
+
+  return isWallThere || isSpriteThere;
+}
+
+function isTileClimbable(x, y) {
+  var tile = bitsy.tile[bitsy.getTile(x, y)];
+  return tile && hackOptions.isClimbable(tile)
+}
+
+function isOnStandableTile(player) {
+  if (fallCounter > 1) return;
+  var coords = [player.x, player.y]
+  var offset = offsets[gravityDir]; // like [0, -1] for y -= 1
+  coords[0] += offset[0]
+  coords[1] += offset[1]
+  var tile = bitsy.tile[bitsy.getTile(coords[0], coords[1])];
+  return tile && hackOptions.isStandable(tile)
+}
+
+function canMoveHorizontallyWhileFalling() {
+  var withinMaxRatio = (horizontalFallingMoves / fallCounter) <= hackOptions.maxHorizontalFallingRatio;
+
+  // if fallCounter is 0 (start of fall/jump) or player last moved down and is within ratio
+  return !fallCounter || (lastMoveMapped === 'down' && withinMaxRatio);
+}
+
+function reallyMovePlayer(player, dir) {
+  if (isSolid(dir, player.x, player.y)) {
+    // can't move into solid thing...so...chill?
+    // should maybe trigger sprites here?
+    return;
+  }
+
+  // why doesn't isSolid catch the out of bounds? stuff? isWall should as well? weird...
+  switch (dir) {
+    case 'up':
+      if (player.y > 0) player.y -= 1;
+      break;
+    case 'down':
+      if (player.y < bitsy.mapsize - 1) player.y += 1;
+      break;
+    case 'left':
+      if (player.x > 0) player.x -= 1;
+      break;
+    case 'right':
+      if (player.x < bitsy.mapsize - 1) player.x += 1;
+      break;
+    default:
+      console.warn('gravity: invalid move', player.x, player.y, dir)
+  }
+}
+
+
+function flipAvatar(gravityDirection) {
+  var hflip = false;
+  var vflip = false;
+  var rot = false;
+  // determine which directions need flipping
+  switch (gravityDirection) {
+  case 'up':
+    vflip = true;
+    break;
+  case 'down':
+    vflip = false;
+    break;
+  case 'left':
+    vflip = true;
+    rot = true;
+    break;
+  case 'right':
+    rot = true;
+    break;
+  default:
+    break;
+  }
+
+  transformAvatar(originalAnimation, vflip, hflip, rot)
+}
+
+addDualDialogTag("setGravityDirection", function (env, params) {
+  var newGravityDir = params[0];
+  if (isValidDir(newGravityDir)) {
+    gravityDir = newGravityDir;
+  } else {
+    console.error('gravity: setGravityDirection failed, expected "up|down|left|right", but got ', newGravityDir);
+  }
+  if (hackOptions.flipAvatarOnGravityChange) flipAvatar(newGravityDir);
+})
+
+addDualDialogTag('toggleGravity', function () {
+  active = !active;
+});
+
+addDualDialogTag('setJumpPower', function (env, params) {
+  var amt = params[0];
+  if (isNaN(amt - 1)) {
```

Comment:
"best practice" way of doing this is:
```suggestion
  if (isNaN(Number(amt))) {
```
what you have would work too but it's working via the implicit cast; the explicit version makes it clearer how/why it works